src/test/java/fr/sncf/osrd/infra_state/RouteStateTest.java

```diff
@@ -550,6 +555,60 @@ public void testCircularInfraReserves() {
         }
     }
 
+    @Test
+    public void testTinyInfra2Trains() throws InvalidInfraException {
+        final var config = TestConfig.readResource("tiny_infra/config_railjson_2trains.json");
+        
+        var infra = config.rjsInfra;
+        var switchID = infra.switches.stream().findFirst().get().id;
+        var trainOrderedList = new ArrayList<String>();
+        trainOrderedList.add("First");
+        trainOrderedList.add("Second");
+
+        var initTable = new RJSSuccessionTable(switchID, trainOrderedList.toArray(new String[2]));
+        config.rjsSuccessions = new RJSSuccessions(Collections.singletonList(initTable));
+
+        var simState = config.prepare();
+        simState.run();
+        var sim = simState.sim;
+        var secondTrain = sim.trains.get("Second");
+        var finalPosition = secondTrain.getLastState().location.getPathPosition();
+        var initialPosition = Train.getInitialLocation(secondTrain.schedule, sim).getPathPosition();
+
+        // Check that the second train moves
+        assertNotEquals(initialPosition, finalPosition);
+    }
+
+    @Test
+    public void testChangeSuccessionTable() throws InvalidInfraException {
+        final var config = TestConfig.readResource("tiny_infra/config_railjson_2trains.json");
+
+        var switchID = config.rjsInfra.switches.stream().findFirst().get().id;
+        var trainOrderedList = new ArrayList<String>();
+        trainOrderedList.add("First");
+        trainOrderedList.add("Second");
+
+        var initTable = new RJSSuccessionTable(switchID, trainOrderedList.toArray(new String[2]));
+        config.rjsSuccessions = new RJSSuccessions(Collections.singletonList(initTable));
+
+        var simState = config.prepare();
+        var sim = simState.sim;
+
+        // Change the succession Table
+        var newTrainOrderedList = new ArrayList<String>();
+        newTrainOrderedList.add("Second");
+        newTrainOrderedList.add("First");
+        var state = sim.infraState.towerState.state.get(switchID);
+        state.table.trainOrderedList = newTrainOrderedList;
```

Comment:
please add a method to towerstate to do this, which let you change all the classes / fields back to private